File: java/src/Solution001.java
import java.util.HashMap;
import java.util.Map;

public class Solution001 {
    public int[] twoSum(int[] nums, int target) {
        int[] indices = new int[2];
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; ++i) {
            if (map.containsKey(nums[i])) {
                indices[0] = i;
                indices[1] = map.get(nums[i]);
                return indices;
            }

            map.put(target-nums[i], i);
        }

        return indices;
    }
}

File: python/Solution001.py
class Solution:
    def twoSum(self, nums, target):
        map = {}
        for i in range(len(nums)):
            if nums[i] in map.keys():
                return [i, map[nums[i]]]
            map[target-nums[i]] = i   # i的键就是i的另一半
        return []


if __name__ == '__main__':
    solution = Solution().twoSum([2, 7, 11, 15], 9)
    print(solution)
